### PR TITLE
Features/save preference type of proof #162422018

### PR DESCRIPTION
**Pull request by akegan**
https://www.pivotaltracker.com/story/show/162422018

### CODE REVIEW

File: app/javascript/components/supplemental_application/sections/preferences/AntiDisplacementHousingPanel.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ export const AntiDisplacementHousingPanel = ({ preferenceIndex, preference }) =>
             TODO: Add ability for users to change application member on pref.
             For now, we just show the current app member in a read-only field.
           */}
-          <input value={memberNameFromPref(preference)} disabled='true' />
```
Comment (akegan) [inline app/javascript/components/supplemental_application/sections/preferences/AntiDisplacementHousingPanel.js:27]: disabled='true' was giving me a console warning
disabled={true} was giving me a linting error on overcommit. 

I got this solution from https://stackoverflow.com/questions/38659884/using-boolean-value-of-attributes-in-jsx

**Comment by akegan** [comment]
so it turns out on this one that it was actually working for preferences other than live/work, but it'll be good to explicitly test for it

**Review by callaghanc — CHANGES_REQUESTED** [review]
good so far, @akegan to add some Live/Work-specific tests as discussed in slack

**Comment by akegan** [comment]
@callaghanc @chrisdolendo I've overhauled the e2e tests for preferences and added some new classes and ids for stuff in order to get it to work. 

It'd be great if you guys could both check this out and let me know what you think

File: app/javascript/components/molecules/ExpandableTable.js [hunk 2]
```diff
@@ -71,12 +74,13 @@ class ExpandableTable extends React.Component {
...
 
-ExpandableTable.ExpanderButton = ({ onClick, label = 'Expand' }) => {
+ExpandableTable.ExpanderButton = ({ onClick, label = 'Expand', id }) => {
   return (
     <button
       type='button'
       className='button button-link action-link'
-      onClick={onClick}>
+      onClick={onClick}
+      id={id !== null ? id : null}>
```
Comment (callaghanc) [inline app/javascript/components/molecules/ExpandableTable.js]: @akegan are you anticipating a case here where `id` would have a falsey value that we wouldn't consider suitable?

Comment (callaghanc) [reply app/javascript/components/molecules/ExpandableTable.js]: as in, if `id` is `false` or `undefined`, do we not consider that suitably falsey here for some reason?

Comment (akegan) [reply app/javascript/components/molecules/ExpandableTable.js]: nope we don't need it, I updated some of the other ones but not this one. I'll remove the check from this one now.

File: spec/javascript/e2e/SupplementalApplicationPage/preferences.e2e.js [hunk 3]
```diff
@@ -7,55 +7,142 @@ import {
...
+    const liveWorkExpandedPanelSelector = '[id$=in-san-francisco-preference-panel]'
 
-    // Click on the first preference's Edit button in the Confirmed
+    // Click on the live/work preference's Edit button in the Confirmed
     // Preferences section to expand that preference's edit panel
-    await page.click(`${firstRowSelector} .button-link`)
+    await page.click(liveWorkEditSelector)
+
+    // Update the individual preference (select Live vs Work in SF)
+    const individualPreferenceSelector = '.individual-preference-select'
```
Comment (callaghanc) [inline spec/javascript/e2e/SupplementalApplicationPage/preferences.e2e.js]: @akegan is there a reason why this selector and the `typeOfProofSelector` are not using `liveWorkExpandedPanelSelector` like the `prefStatusSelector`?

File: spec/javascript/e2e/SupplementalApplicationPage/preferences.e2e.js [hunk 4]
```diff
@@ -7,55 +7,142 @@ import {
...
+    const currentTypeOfProof = await page.$eval(getSelectedOptSelector(typeOfProofSelector), e => e.textContent)
+    expect(currentTypeOfProof).toBe(typeOfProofToSetName)
+
+    const currentStatus = await page.$eval(getSelectedOptSelector(prefStatusSelector), e => e.textContent)
+    expect(currentStatus).toBe(prefStatusToSetName)
+
+    await browser.close()
+  }, DEFAULT_E2E_TIME_OUT)
+
+  test('should allow updates to assisted housing preference', async () => {
```
Comment (callaghanc) [inline spec/javascript/e2e/SupplementalApplicationPage/preferences.e2e.js:96]: @akegan looks like the comments in this test need to be updated from referencing live/work to reference the assisted housing pref

**Review by callaghanc — CHANGES_REQUESTED** [review]
thank you @akegan for expanding our confirmed pref e2e tests! great work, just a couple small details to resolve and then this is good to go

**Comment by akegan** [comment]
tests have (finally) passed, so this is ready for review again @callaghanc

**Review by callaghanc — DISMISSED** [review]

**Comment by akegan** [comment]
@chrisdolendo I just requested you for review on this. Catherine had approved, but I ended up making a pretty major change in response to a bug. 

For more details check out the last comment on the ticket: https://www.pivotaltracker.com/story/show/162422018

(Still working on getting tests to pass)

**Comment by akegan** [comment]
e2e tests unearthed an interesting bug: TBD whether or not I can fix it easily in this PR https://www.pivotaltracker.com/story/show/161691130

**Comment by akegan** [comment]
@chrisdolendo my grand plan to get rid of all references to lwPreferenceProof appears to have failed unfortunately. It seems like for the direct preference update we do have to use LW_Type_of_Proof

**Comment by akegan** [comment]
@chrisdolendo  this is finally ready for review again. the logic here is a bit complicated, so please do QA before we hand it off for product acceptance

**Comment by akegan** [comment]
@chrisdolendo this one is still waiting for review FYI

**Comment by akegan** [comment]
@callaghanc if you have time it might make more sense for you to review this again.

It looks like I may have messed up the commit history when I rebased to remove my attempt at totally removing lwPreference proof. There's a lot more context into the changes in the pivotal ticket https://www.pivotaltracker.com/story/show/162422018

**Comment by callaghanc** [comment]
@akegan sounds good, i will take a look when i get to catching up on github. have made it through slack and email catch-ups so far, and am currently adventuring through pivotal!

**Comment by callaghanc** [comment]
@akegan above you said:
> e2e tests unearthed an interesting bug: TBD whether or not I can fix it easily in this PR https://www.pivotaltracker.com/story/show/161691130

Looking at that story it's titled "Hide test listings in lease-up from webapp" - could you explain how it's related to this PR?

**Comment by akegan** [comment]
It must have been a typo! here's the bug that I think I was trying to refer to https://www.pivotaltracker.com/story/show/162748715

**Review by callaghanc — APPROVED** [review]

**Pull request merged**